common/djangoapps/course_modes/admin.py

```diff
@@ -29,12 +29,15 @@
 from lms.djangoapps.verify_student import models as verification_models
 
 
-class CourseModeForm(forms.ModelForm):
-
-    class Meta(object):
-        model = CourseMode
-        fields = '__all__'
+COURSE_MODE_SLUG_CHOICES = []
+try:
+    COURSE_MODE_SLUG_CHOICES = [(mode.mode_slug, mode.mode_slug) for mode in EnrollmentMode.objects.all()]
+except:  # pylint: disable=bare-except
+    pass
 
+if not COURSE_MODE_SLUG_CHOICES:
+    # The entries in this table are populated in a migration file, and migration files do not
```

Comment:
This is sad-- I was hoping to delete the hardcoded COURSE_MODE_SLUG_CHOICES, but can't because of unit tests.